relay-dynamic-config/src/project.rs

```diff
@@ -90,6 +93,35 @@ impl ProjectConfig {
         metrics::convert_conditional_tagging(self);
         defaults::add_span_metrics(self);
     }
+
+    /// Combines the measurementsconfig from global config and project config.
+    pub fn measurements<'a>(
+        &'a self,
+        global_config: &'a Arc<GlobalConfig>,
+    ) -> Option<MeasurementsConfig> {
+        match (
+            self.measurements.as_ref(),
+            global_config.measurements.as_ref(),
+        ) {
+            (None, None) => None,
+            (None, Some(glob_measurements_config)) => Some(glob_measurements_config.clone()),
```

Comment:
Is this `clone` allocating new memory? The goal of having this accessor method is to provide a way to access measurements from both project and global configs without allocating new memory.

Reply:
yep, it is! I wrote my thoughts on it here: https://github.com/getsentry/relay/pull/2415#discussion_r1308348181   
 
I'll make it now without new allocations, so we can decide how to balance performance vs simplicity

Reply:
these lifetimes issues driving me crazy